src/uu/du/src/du.rs

```diff
@@ -306,49 +320,69 @@ fn du(
             }
         };
 
-        for f in read {
+        'file_loop: for f in read {
             match f {
-                Ok(entry) => match Stat::new(entry.path(), options) {
-                    Ok(this_stat) => {
-                        if let Some(inode) = this_stat.inode {
-                            if inodes.contains(&inode) {
-                                continue;
-                            }
-                            inodes.insert(inode);
-                        }
-                        if this_stat.is_dir {
-                            if options.one_file_system {
-                                if let (Some(this_inode), Some(my_inode)) =
-                                    (this_stat.inode, my_stat.inode)
-                                {
-                                    if this_inode.dev_id != my_inode.dev_id {
-                                        continue;
+                Ok(entry) => {
+                    match Stat::new(entry.path(), options) {
+                        Ok(this_stat) => {
+                            if !&exclude.is_empty() {
+                                // We have an exclude list
+                                for pattern in exclude {
+                                    // Look at all patterns
+                                    if pattern.matches(&this_stat.path.to_string_lossy())
+                                        || pattern
+                                            .matches(&entry.file_name().into_string().unwrap())
```

Comment:
Could you explain this condition? Is it GNU behaviour that both are allowed to match? Also, can we use `entry.path()` and do the check before the `Stat::new()`, so that we don't do additional syscalls? Or do we have to know whether `Stat::new` succeeds?

Reply:
this_stat.path returns the full path, needed for a/b search
the entry.file_name(), just the filename 

Both are needed

Reply:
More details:

if you have 
mkdir terts/diepraam/
and do:
du terts

entry will be "terts"
but you can do
du --exclude terts/diepraam

So, you need to ignore 
terts/diepraam 
so, you need the full path too